modules/ob-authentication-to-container-registries.adoc

```diff
@@ -0,0 +1,47 @@
+// This module is included in the following assembly:
+//
+// builds/authenticating-build-images.adoc
+
+:_content-type: PROCEDURE
+[id="ob-authentication-to-container-registries_{context}"]
+= Authentication to container registries
+
+To push images to a private container registry, you must define a secret in the `OpenShift-Builds` namespace and then reference it in your `Build` custom resource (CR).
+
+
+.Procedure
+. Run the following command to generate a secret:
++
+[source,terminal]
+----
+oc --namespace <namespace> create secret docker-registry <container_registry_secret_name> \
+  --docker-server=<registry_host> \ <1>
+  --docker-username=<username> \ <2>
+  --docker-password=<password> \ <3>
+  --docker-email=<email_address>
+---- 
+<1> The `<REGISTRY_HOST>` value must be set to `\https://index.docker.io/v1/`
+<2> The `<USERNAME>` is the Docker ID
+<3> The `<PASSWORD>` value can be your docker hub password or an access token
```

Comment:
```suggestion
<1> The `<REGISTRY_HOST>` value must be set to `\https://index.docker.io/v1/`.
<2> The `<USERNAME>` is the Docker ID.
<3> The `<PASSWORD>` value can be your docker hub password or an access token.
```

Reply:
Probably out of scope for a peer/merge review, but is this procedure supposed to be specifically about Docker Hub authentication? Or just generally about authentication to any private container registry (similar to what's described in https://docs.openshift.com/container-platform/4.13/openshift_images/image-streams-manage.html#images-allow-pods-to-reference-images-from-secure-registries_image-streams-managing, for example)? I was surprised to see `<registry_host>` had to be `index.docker.io` and references to "docker hub" (which I think also should be "Docker Hub", if used).

Reply:
@apoorvajagtap Could you please help with this comment?

Reply:
@shipsing this could be generic.

Reply:
To make it generic, we need to update the example as well. because we mentioned docker-server, docker-username, docker-password and other parameters as well. We have asked the user to set the registry host to https://index.docker.io/v1/ (which again makes this example specific to docker hub).

@apoorvajagtap Could you please share a generic example with me in this context?

Reply:
@shipsing with making it generic we do not need to modify anything in the command. Actually kubernetes/openshift have secrets of 3 types, and `docker-registry` is one of the type. So, even if we have to configure a secret for another registry, the parameters will remain same (this was due to the usage of docker instead of container earlier).